lib/buildkite/builder/pipeline.rb

```diff
@@ -140,20 +140,12 @@ def template(name, &definition)
         @templates[name.to_s] = definition
       end
 
-      def processors(*processor_classes)
-        unless processor_classes.empty?
-          @processors.clear
-
-          processor_classes.flatten.each do |processor|
-            unless processor < Buildkite::Builder::Processors::Abstract
-              raise "#{processor} must inherit from Buildkite::Builder::Processors::Abstract"
-            end
-
-            @processors << processor
-          end
+      def use(processor, **args)
+        unless processor < Buildkite::Builder::Processors::Abstract
+          raise "#{processor} must inherit from Buildkite::Builder::Processors::Abstract"
         end
 
-        @processors
+        @processors << processor.new(self, **args)
```

Comment:
Hm, I just realized that for processors to take advantage of `args`, it'd have to overwrite `initialize`. Let's get around this by holding on the the `args` and passing them into `Processor#run`, which it can be forwarded on to `Processor#process`.